Common/Utils/include/CommonUtils/BoostHistogramUtils.h

```diff
@@ -270,41 +322,55 @@ fitResult<T, 4> fitGaus(Iterator first, Iterator last, AxisIterator axisfirst)
     }
 
     T param1 = T(par[1] / (-2. * par[2]));
-    result.setParameter<1>(param1);
-    result.setParameter<2>(T(1. / TMath::Sqrt(TMath::Abs(-2. * par[2]))));
+    //result.setParameter<1>(param1);
+    //result.setParameter<2>(T(1. / TMath::Sqrt(TMath::Abs(-2. * par[2]))));
+    result.at(1) = param1;
+    result.at(2) = T(1. / TMath::Sqrt(TMath::Abs(-2. * par[2])));
     auto lnparam0 = par[0] + par[1] * param1 + par[2] * param1 * param1;
     if (lnparam0 > 307) {
       throw FitGausError_t::FIT_ERROR;
       ;
     }
-    result.setParameter<0>(TMath::Exp(lnparam0));
-
+    //result.setParameter<0>(TMath::Exp(lnparam0));
+    result.at(0) = TMath::Exp(lnparam0);
     return result;
   }
 
   if (npoints == 2) {
     //use center of gravity for 2 points
     meanCOG /= sumCOG;
     rms2COG /= sumCOG;
+    /*
     result.setParameter<0>(xMax);
     result.setParameter<1>(meanCOG);
     result.setParameter<2>(TMath::Sqrt(TMath::Abs(meanCOG * meanCOG - rms2COG)));
     result.setChi2(-2);
+    */
+    result.at(0) = *xMax;
+    result.at(1) = meanCOG;
+    result.at(2) = TMath::Sqrt(TMath::Abs(meanCOG * meanCOG - rms2COG));
```

Comment:
This looks like a template function. In this case it would really be better to use `std::` math functions since they are also templates. The ROOT ones are just `double` I think and there is no reason preferring ROOT math functions over the standard (for things such as exp, sin, cos,...).

Reply:
Maybe this could be cleaned up in a separate PR